### PR TITLE
Features/wfcordinator sdra open assessments

### PR DESCRIPTION
**Pull request by benbhall**
- Add Common.Helpers project
- Add Common.Messages project
- Update Microsoft.Extensions.Configuration.AzureAppConfiguration package
- Tweak Process EF Model to use Table data attribute so that the table can be mapped
- Tweak SQL connection strings
- Solution-wide adjustment of namespaces
- Add AssessmentPollingSaga
- Add AssessmentPollingSagaTests
- Add DataServiceApiClient
- Add StartDbAssessmentCommandHandler
- Rework config POCOs

### CODE REVIEW

File: src/WorkflowCoordinator/WorkflowCoordinator/NServiceBusJobHost.cs [hunk 1]
```diff
@@ -1,83 +1,180 @@
...
+                    .DefiningCommandsAs(type => type.Namespace == "Common.Messages.Commands")
+                    .DefiningEventsAs(type => type.Namespace == "Common.Messages.Events")
+                    .DefiningMessagesAs(type => type.Namespace == "Common.Messages")
+                    .DefiningMessagesAs(type => type.Namespace == "WorkflowCoordinator.Messages");
+                ;
+                endpointConfiguration.AssemblyScanner().ScanAssembliesInNestedDirectories = true;
+                endpointConfiguration.EnableInstallers();
                 _endpoint = await Endpoint.Start(endpointConfiguration);
+
+                Guid pollingSagaGuid = new Guid("ef3932c3-4232-41be-a0e7-05cf8a052017");
```
Comment (UKHOGreg) [inline src/WorkflowCoordinator/WorkflowCoordinator/NServiceBusJobHost.cs]: Guid should be retrieved via config

File: src/WorkflowCoordinator/WorkflowCoordinator/Program.cs [hunk 2]
```diff
@@ -19,45 +26,79 @@ private static async Task Main(string[] args)
...
+                hostingContext.Configuration.GetSection("nsb").Bind(startupConfig);
+                hostingContext.Configuration.GetSection("databases").Bind(startupConfig);
+
+                var endpointConfiguration = new EndpointConfiguration(startupConfig.WorkflowCoordinatorName);
+
+                services.AddSingleton<EndpointConfiguration>(endpointConfiguration);
+                services.AddOptions<GeneralConfig>()
+                    .Bind(hostingContext.Configuration.GetSection("nsb"))
+                    .Bind(hostingContext.Configuration.GetSection("apis"))
+                    .Bind(hostingContext.Configuration.GetSection("urls"));
```
Comment (UKHOGreg) [inline src/WorkflowCoordinator/WorkflowCoordinator/Program.cs]: We need to bind against databases section to get the LocalDbServer

File: src/WorkflowCoordinator/WorkflowCoordinator/NServiceBusJobHost.cs [hunk 3]
```diff
@@ -1,83 +1,180 @@
...
             _secretsConfig = secretsConfig;
+            _hostingEnvironment = hostingEnvironment;
+            _endpointConfig = endpointConfig;
+
+            _isLocalDebugging = _hostingEnvironment.IsDevelopment() && Debugger.IsAttached;
+            _localDbServer = _generalConfig.Value.LocalDbServer;
+
+            if (_isLocalDebugging)
+            {
+                _connectionString = DatabasesHelpers.BuildSqlConnectionString(_isLocalDebugging, _secretsConfig.Value.NsbInitialCatalog);
```
Comment (UKHOGreg) [inline src/WorkflowCoordinator/WorkflowCoordinator/NServiceBusJobHost.cs]: We need to pass the dataSource (DB Server)

File: src/Databases/WorkflowDatabase.Tests/ExampleTests.cs [hunk 4]
```diff
@@ -14,7 +15,7 @@ public class ExampleTests
         public void Setup()
         {
             var dbContextOptions = new DbContextOptionsBuilder<WorkflowDbContext>()
-                .UseSqlServer(@"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=WorkflowDatabase;Integrated Security=True;Persist Security Info=False;Pooling=False;MultipleActiveResultSets=False;Connect Timeout=60;Encrypt=False;TrustServerCertificate=False")
+                .UseSqlServer(@"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=taskmanager-dev-workflowdatabase;Integrated Security=True;Persist Security Info=False;Pooling=False;MultipleActiveResultSets=False;Connect Timeout=60;Encrypt=False;TrustServerCertificate=False")
```
Comment (Rossall82) [inline src/Databases/WorkflowDatabase.Tests/ExampleTests.cs]: Should we pull this from config?

Comment (UKHOGreg) [reply src/Databases/WorkflowDatabase.Tests/ExampleTests.cs]: We shall accept this as it is in a unit test and we do not have config integration there.

File: src/WorkflowCoordinator/WorkflowCoordinator/Models/AssessmentModel.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using Newtonsoft.Json;
+
+namespace WorkflowCoordinator.Models
+{
+    public class AssessmentModel
```
[the remark this answers is not in the document]
Comment (UKHOGreg) [reply src/WorkflowCoordinator/WorkflowCoordinator/Models/AssessmentModel.cs]: Team has agreed to make this change on a separate branch.

**Pull request merged**